modules/groovy-dsl/Assertions.md

```diff
@@ -0,0 +1,232 @@
+# Assertions
+
+Until you've added assertions to your DSL script, you're not really testing your application. You're just checking that 
+it doesn't blow up in your face when you access it.
+
+If you do add assertions, steps in your test will be marked as FAILED if the assertions fail. If not, they will have status
+ERROR if the request itself fails (e.g. because of network connectivity issues), but if a response is received the 
+status will be set to UNKNOWN. If no steps have the status FAILED or ERROR, the test will automatically be considered
+successful.
+
+Assertions are always added in an ```asserting``` section right after the request step. See the assertion descriptions below 
+for examples.
+
```

Comment:
could we insert a (manually curated) TOC here?

Reply:
Probably, although Github MD has some problems with generating anchors for some of the headers.